report/renderer/topbar-features.js

```diff
@@ -273,20 +254,39 @@ export class TopbarFeatures {
     }
   }
 
-  _setupStickyHeaderElements() {
+  _setupStickyHeader() {
+    // Cache these elements to avoid qSA on each onscroll.
     this.topbarEl = this._dom.find('div.lh-topbar', this._document);
-    this.scoreScaleEl = this._dom.find('div.lh-scorescale', this._document);
-    this.stickyHeaderEl = this._dom.find('div.lh-sticky-header', this._document);
+    this.categoriesEl = this._dom.find('div.lh-categories', this._document);
+
+    // Only present in the DOM if it'll be used (>=2 categories)
+    try {
+      this.stickyHeaderEl = this._dom.find('div.lh-sticky-header', this._document);
+    } catch (e) {}
+
+    if (!this.stickyHeaderEl) return;
 
     // Highlighter will be absolutely positioned at first gauge, then transformed on scroll.
     this.highlightEl = this._dom.createChildOf(this.stickyHeaderEl, 'div', 'lh-highlighter');
+
+    // Defer behind rAF to avoid forcing layout
+    requestAnimationFrame(() => requestAnimationFrame(() => {
+      const scrollParent = this._getScrollParent(this.stickyHeaderEl);
+      scrollParent.addEventListener('scroll', e => this._updateStickyHeader(e));
+      // 'scroll' handler must be Element | Document, but resizeObserve can't be Document.
+      new window.ResizeObserver(e => this._updateStickyHeader(e))
```

Comment:
nit: Can we bind `this._updateStickyHeader` instead? Also, looks tike there is no need to pass the `e` through.

Reply:
> nit: Can we bind `this._updateStickyHeader` instead?

we could. these two lines would look like this (or something close):

```js
scrollParent.addEventListener('scroll', this._updateStickyHeader.bind(this));
new window.ResizeObserver(this._updateStickyHeader.bind(this))
```

i personally dont like the binding style because you create a new function object which has a separate identity from the actual implementation.

so ever since arrow functions have been a thing, i think it makes for a cleaner look without such indirection.


> Also, looks tike there is no need to pass the `e` through.

true! good call. coulda just done a `() => this.USH()`. thats def better